### ISSUE TITLE
In the language model  transformer_lm_wiki103 , the code of decoder attention heads=8 which is different from paper

### ISSUE THREAD

**Issue by zhilizju** [pytorch/fairseq#2612]
In the language model  [transformer_lm_wiki103](https://github.com/pytorch/fairseq/blob/master/fairseq/models/transformer_lm.py) , the code of decoder attention heads=8 which is different from paper [Baevski and Auli, 2018](https://arxiv.org/abs/1809.10853) 16

### code
`def transformer_lm_baevski_wiki103(args):
    args.decoder_layers = getattr(args, 'decoder_layers', 16)
    args.decoder_attention_heads = getattr(args, 'decoder_attention_heads', 8)`
### paper
> Each block contains two sub-blocks: the first is a multihead self-attention module with H = 16 heads

which one is right? I  followed the [instruction](https://github.com/pytorch/fairseq/blob/master/examples/language_model/README.adaptive_inputs.md) and got the result

`2020-09-14 11:02:59 | INFO | fairseq_cli.eval_lm | Loss (base 2): 4.2494, Perplexity: 19.02` 
which is higher than paper **18.7**   Is that because of the number of attention heads?

I have four GPUs listed  as follows:

`2020-09-01 10:14:02 | INFO | fairseq.utils | ***********************CUDA enviroments for all 4 workers***********************
2020-09-01 10:14:02 | INFO | fairseq.utils | rank   0: capabilities =  7.5  ; total memory = 47.462 GB ; name = Quadro RTX 8000
2020-09-01 10:14:02 | INFO | fairseq.utils | rank   1: capabilities =  7.5  ; total memory = 47.462 GB ; name = Quadro RTX 8000
2020-09-01 10:14:02 | INFO | fairseq.utils | rank   2: capabilities =  7.5  ; total memory = 47.462 GB ; name = Quadro RTX 8000
2020-09-01 10:14:02 | INFO | fairseq.utils | rank   3: capabilities =  7.5  ; total memory = 47.462 GB ; name = Quadro RTX 8000
`
Here is my command:
`python train.py --task language_modeling data-bin/wikitext-103 --save-dir checkpoints/transformer_wikitext-103 --arch transformer_lm_wiki103 --max-update 286000 --max-lr 1.0 --t-mult 2 --lr-period-updates 270000 --lr-scheduler cosine --lr-shrink 0.75 --warmup-updates 16000 --warmup-init-lr 1e-07 --min-lr 1e-09 --optimizer nag --lr 0.0001 --clip-norm 0.1 --criterion adaptive_loss --max-tokens 4096 --update-freq 4 --tokens-per-sample 3072 --seed 1 --sample-break-mode none --skip-invalid-size-inputs-valid-test --ddp-backend=no_c10d`

I set `--max-tokens 4096 --update-freq 4 `  to keep batch size is the same as [paper](https://arxiv.org/abs/1809.10853) 

Thanks a lot

**Comment by zhilizju** [comment]
I have another question.
How to reproduce the base model in the paper (Baevski and Auli, [2018)](https://arxiv.org/abs/1809.10853) 

> We consider a word-based setup with fixed size word input embeddings and a standard word softmax
(SM) where embeddings have either dimension 512 (WIKITEXT-103)

 **

> model_name  Input Output Valid Test Train Time(hours) Params
> 
>  SM  Embedding  Softmax 23.87 24.92 57* 476.8M`

**

It seems that the hyperparameters of [@register_model_architecture('transformer_lm', 'transformer_lm')](https://github.com/pytorch/fairseq/blob/master/fairseq/models/transformer_lm.py) is different from the paper.
Should I register a new model myself?  Can anyone provide me some help?

**Comment by lematt1991** [comment]
CC @alexeib

**Comment by zhilizju** [comment]
Can you give me some help? @alexeib

**Comment by alexeib** [comment]
you can define a new model by using some base architecture and overwriting parameters via command line flags (config file support coming soon!). we dont have a built in architecture for baseline softmax word models
